private/buf/cmd/buf/command/breaking/breaking.go

```diff
@@ -339,3 +351,103 @@ func validateFlags(flags *flags) error {
 	}
 	return nil
 }
+
+// hasNoUniqueBreakingConfig iterates through imageWithConfigs and checks to see if there
+// are any unique [bufconfig.BreakingConfig]. It returns true if all [bufconfig.BreakingConfig]
+// are the same across all the images.
+func hasNoUniqueBreakingConfig(imageWithConfigs []bufctl.ImageWithConfig) bool {
+	var base bufconfig.BreakingConfig
+	for _, imageWithConfig := range imageWithConfigs {
+		if base == nil {
+			base = imageWithConfig.BreakingConfig()
+			continue
+		}
+		if !equalBreakingConfig(base, imageWithConfig.BreakingConfig()) {
+			return false
+		}
+		base = imageWithConfig.BreakingConfig()
+	}
+	return true
+}
+
+// Checks if the specified [bufconfig.BreakingConfig]s are equal. Returns true if both
+// [bufconfig.BreakingConfig] have the same configuration parameters.
+func equalBreakingConfig(breakingConfig1, breakingConfig2 bufconfig.BreakingConfig) bool {
+	if breakingConfig1.Disabled() != breakingConfig2.Disabled() {
+		return false
+	}
+	if breakingConfig1.FileVersion() != breakingConfig2.FileVersion() {
+		return false
+	}
+	if slices.Compare(breakingConfig1.UseIDsAndCategories(), breakingConfig2.UseIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.ExceptIDsAndCategories(), breakingConfig2.ExceptIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.IgnorePaths(), breakingConfig2.IgnorePaths()) != 0 {
+		return false
+	}
+	for ignoreID, paths1 := range breakingConfig1.IgnoreIDOrCategoryToPaths() {
+		paths2, ok := breakingConfig2.IgnoreIDOrCategoryToPaths()[ignoreID]
+		if !ok {
+			return false
+		}
+		if slices.Compare(paths1, paths2) != 0 {
+			return false
+		}
+	}
+	if breakingConfig1.DisableBuiltin() != breakingConfig2.DisableBuiltin() {
+		return false
+	}
+	return breakingConfig1.IgnoreUnstablePackages() == breakingConfig2.IgnoreUnstablePackages()
+}
+
+// A helper function for filtering out images from [imagesWithConfig] if there is no corresponding
+// image in [againstImages]. We determine this based on image file path.
+//
+// This assumes that len(imageWithConfigs) > len(againstImages).
+// We also expect that each image in [againstImages] is mapped only once to a single
+// imageWithConfig in [imagesWithConfig]. If an againstImage is found, then we don't check
+// it again. We also validate that each image in [againstImages] is mapped to an imageWithConfig
+// from [imageWithConfigs].
+func filterImagesNotInAgainstImages(
+	imageWithConfigs []bufctl.ImageWithConfig,
+	againstImages []bufimage.Image,
+) ([]bufctl.ImageWithConfig, error) {
+	foundAgainstImageIndices := make(map[int]struct{})
+	var filteredImageWithConfigs []bufctl.ImageWithConfig
+	for _, imageWithConfig := range imageWithConfigs {
+		for _, imageFile := range imageWithConfig.Files() {
```

Comment:
This is stable per invocation, but could change depending on the DAG order of the files. For example, if the first file matches against A, but the second file matches against B. Do we need to exhaustively check that all the against paths match the same image or no image at all (deleted)?